MaxText/tests/fwd_pass_testutil.py

```diff
@@ -0,0 +1,105 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import sys
+import os
+import pytest
+import unittest
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+maxtext_parent_dir = os.path.dirname(current_dir)
+print(f"Adding parent directory = {maxtext_parent_dir}")
+sys.path.append(maxtext_parent_dir)
+
+import common_types
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+import pyconfig
+
+import jsonlines
+import train
+
+
+def get_data(golden_data, golden_data_index, config):
+    print(f"Comparing forward pass for golden data index = {golden_data_index} ")
+    print(f"config.global_batch_size_to_train_on={config.global_batch_size_to_train_on}")
+    s = (config.global_batch_size_to_train_on, config.max_target_length)
+    ids = np.asarray(golden_data[golden_data_index]['tokens'], dtype=np.int32)
+    
+    logits = np.asarray(golden_data[golden_data_index]['logits'], dtype=np.float32)
+    print(f" prompt=\"{golden_data[golden_data_index]['prompt']}\" raw ids={ids}, logits.shape = {logits.shape}")
+
+
+    decoder_segment_ids = jax.numpy.zeros(s) + common_types.DECODING_ACTIVE_SEQUENCE_INDICATOR
+    decoder_positions = jnp.stack(
+        [jnp.arange(config.max_target_length, dtype=jnp.int32) for _ in range(config.global_batch_size_to_train_on)]
+    )
+
+    ids = jnp.stack(
+        [ids for _ in range(config.global_batch_size_to_train_on)]
+    )
+    print(f"ids={ids}, decoder_segment_ids = {decoder_segment_ids}, decoder_positions= {decoder_positions}")
+
+    return ids, decoder_segment_ids, decoder_positions, logits
+
+def main(config):
+  """Test the Whole Model of model_name"""
+
+  #initialize the Llama2-7b model with weights from Meta
+  (
+      init_rng,
+      writer,
+      checkpoint_manager,
+      state_mesh_annotations,
+      model,
+      mesh,
+      learning_rate_schedule,
+      data_iterator,
+      eval_data_iterator,
+      state,
+    ) = train.setup_train_loop(config)
+  
+  input_golden_data_path = "MaxText/test_assets/golden_data_"+config.model_name+".jsonl"  
```

Comment:
probably worth exposing directly?

Reply:
like kind of weird to be guessing -- what if the someone wants to test multiple prompts, etc.

Reply:
>probably worth exposing directly?

Since we are comparing logts, for Llama2 we are looking for `[sequence_length,32000]` long logits, it would be hard to expose directly in this `.py` file itself.

This input `.jsonl` file has multiple prompts already, three actually, users can add more.

Is this what you meant?